model/src/main/kotlin/OrtResult.kt

```diff
@@ -311,7 +319,27 @@ data class OrtResult(
      * Return true if the project or package with the given identifier is excluded.
      */
     @Suppress("UNUSED") // This is intended to be mostly used via scripting.
-    fun isExcluded(id: Identifier): Boolean = getExcludes().isExcluded(id, this)
+    fun isExcluded(id: Identifier): Boolean =
+        getProjects().find { it.id == id }?.let {
+            // An excluded project could still be included as a dependency of another non-excluded project.
+            isProjectExcluded(it) && isPackageExcluded(id)
+        } ?: isPackageExcluded(id)
+
```

Comment:
Nit: Consecutive blank lines. Maybe you can fix this retrospectively, @fviernau?